pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -180,14 +185,9 @@ def on_advance_end(self):
         # update plateau LR scheduler after metrics are logged
         self.update_lr_schedulers("step", update_plateau_schedulers=True)
 
-        self.total_batch_idx += 1
```

Comment:
careful removing this, the call to `_increment_accumulated_grad_global_step` requires the updated value here.

Reply:
@carmocca

Reply:
Fixed it by using `current.ready` instead of `total_batch_idx`.

Not sure why that was the case. Especially since that's different to the check done in `_should_accumulate` which does use `current.ready`.

Long term, I have opened #8856 to address these inconsistencies (WIP)

Thnaks for the eagle eye @awaelchli !